airflow/providers/amazon/CHANGELOG.rst

```diff
@@ -26,6 +26,19 @@
 Changelog
 ---------
 
+Main
+......
+
+.. warning:: A bug in ``RedshiftDataOperator`` prevented tasks configured with ``deferrable=True`` and
+  ``wait_for_completion=True`` from entering the deferred state. Instead, the task would occupy an executor slot
```

Comment:
Do we know which version introduced this bug? If we can be specific that is helpful

Reply:
Version 8.17.0 introduced it (https://github.com/apache/airflow/pull/36586).
I didn't specify it because before that version, the operator was not deferrable so this bug couldn't exist.
But I'll rephrase the warning to still mention the version.